Roles/(Ghosts)/Impostor/Possessor.cs

```diff
@@ -0,0 +1,167 @@
+        .SetValueFormat(OptionFormat.Seconds);
+        AlertRange = FloatOptionItem.Create(Id + 12, "PossessorAlertRange", new(1f, 10f, 0.5f), 2.5f, TabGroup.ImpostorRoles, false).SetParent(CustomRoleSpawnChances[CustomRoles.Possessor])
+        .SetValueFormat(OptionFormat.Multiplier);
+        FocusRange = FloatOptionItem.Create(Id + 13, "PossessorFocusRange", new(5f, 25f, 2.5f), 10f, TabGroup.ImpostorRoles, false).SetParent(CustomRoleSpawnChances[CustomRoles.Possessor])
+        .SetValueFormat(OptionFormat.Multiplier);
+    }
+    public override void Init()
+    {
+    }
+    public override void Add(byte PlayerId)
+    {
+        CustomRoleManager.OnFixedUpdateOthers.Add(OnFixedUpdateOther);
+    }
+    // EAC bans players when GA uses sabotage
+    public override bool CanUseSabotage(PlayerControl pc) => false;
+    public override void ApplyGameOptions(IGameOptions opt, byte playerId)
+    {
+        AURoleOptions.GuardianAngelCooldown = controllingPlayer ? 0f : PossessCooldown.GetFloat();
+        AURoleOptions.ProtectionDurationSeconds = 0f;
+    }
+
+    private void OnFixedUpdateOther(PlayerControl target)
+    {
+        if (target.PlayerId == controllingTargetId)
+        {
+            if (controllingPlayer && possessTime >= 0)
+            {
+                if (CheckRange(_Player.GetCustomPosition(), target.GetCustomPosition()) > 5f)
+                {
+                    _Player.RpcTeleport((_Player.GetCustomPosition() + target.GetCustomPosition()) / 2);
+                }
+
+                foreach (var allPlayers in Main.AllAlivePlayerControls.Where(pc => pc != target))
+                {
+                    if (CheckRange(target.GetCustomPosition(), allPlayers.GetCustomPosition()) < AlertRange.GetFloat())
+                    {
+                        controllingPlayer = false;
+                    }
+                }
+
+                if (CheckRange(_Player.GetCustomPosition(), target.GetCustomPosition()) < 1f)
+                {
+                    if (target.MyPhysics.Animations.IsPlayingRunAnimation())
+                    {
+                        target.RpcTeleport(target.GetCustomPosition());
+                        target.MyPhysics.RpcCancelPet();
+                    }
+                }
+                else if (CheckRange(_Player.GetCustomPosition(), target.GetCustomPosition()) < 3.5f)
+                {
+                    if (!target.petting && CheckRange(_Player.GetCustomPosition(), target.GetCustomPosition()) > 1f)
+                    {
+                        target.MyPhysics.RpcPet(_Player.GetCustomPosition(), new Vector2(500f, 500f));
+                    }
+                    else if (!target.MyPhysics.Animations.IsPlayingRunAnimation())
+                    {
+                        target.MyPhysics.RpcCancelPet();
+                        target.RpcTeleport(target.GetCustomPosition());
+                    }
+                }

```

Comment:
I don't quite understand what this code is

Why does the player need to go to coordinates x 500 y 500

Reply:
He manipulates Petrpc to make the player walk smoothly towards the direction, instead of spamming teleport on the player https://discord.com/channels/1094344790910455908/1270477748443283517/1270477748443283517